cmake/Modules/FindARM.cmake

```diff
@@ -41,7 +41,7 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
    ENDIF (OMAP4_TRUE)
 
 ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Darwin")
-   IF(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
+   IF(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64" AND NOT CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
```

Comment:
Do we have to worry about CMAKE_OSX_ARCHITECTURES equalling any other architecture or nah?

Reply:
In theory we should, but let's do it one step at a time: so far we can only use the system to cross-build it for another platform.